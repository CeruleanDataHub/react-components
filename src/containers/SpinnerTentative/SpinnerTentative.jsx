import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const SpinnerContainer = styled.div`
  margin: 1em;
  width: 60px;
  height: 60px;
  background-color: lightgray;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
`;

const Spinner = styled.div`
  width: 32px;
  height: 32px;
  margin: 8px;
  border: 4px solid #fff;
  border-radius: 50%;
  animation: spinner 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: #fff transparent transparent transparent;

  @keyframes spinner {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
`;

const SpinnerTentative = ({ condition, children }) => {
  return (
    <>
      {condition ? (
        <div data-test="content">{children}</div>
      ) : (
        <SpinnerContainer>
          <Spinner />
        </SpinnerContainer>
      )}
    </>
  );
};

SpinnerTentative.propTypes = {
  condition: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired
};

export default SpinnerTentative;
