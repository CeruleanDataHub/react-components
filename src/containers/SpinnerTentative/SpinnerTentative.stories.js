import { withKnobs } from "@storybook/addon-knobs";
import React, { useEffect, useState } from "react";
import SpinnerTentative from "./SpinnerTentative";

export default {
  title: "SpinnerTentative",
  component: SpinnerTentative,
  decorators: [withKnobs]
};

export const SpinnerTentativeStory = () => {
  return (
    <div style={{ minHeight: 250, width: 300 }}>
      <SpinnerTentative condition={false}>Irrelevant</SpinnerTentative>
    </div>
  );
};

export const ResolveSpinnerTentativeStory = () => {
  const [condition, setCondition] = useState(false);

  useEffect(() => {
    setTimeout(() => {
      setCondition(true);
    }, 5000);
  });

  return (
    <div style={{ minHeight: 250, width: 300 }}>
      <SpinnerTentative condition={condition}>
        This is shown after condition resolves
      </SpinnerTentative>
    </div>
  );
};
