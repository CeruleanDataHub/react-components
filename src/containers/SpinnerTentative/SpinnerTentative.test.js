import React from "react";
import { mount } from "enzyme";
import SpinnerTentative from "./SpinnerTentative";

describe("SpinnerTentative", () => {
  let component;
  let conditionToBeResolved;

  beforeEach(() => {
    conditionToBeResolved = false;
  });

  it("renders", () => {
    component = mount(
      <SpinnerTentative condition={conditionToBeResolved}>
        Some content
      </SpinnerTentative>
    );

    expect(component).toMatchSnapshot();
  });

  it("when condition has not resolved, does not view content", () => {
    const content = component.find("div[data-test='content']");
    expect(content).not.toExist();
  });

  it("when condition resolves, views content", () => {
    component.setProps({ condition: true });

    const content = component.find("div[data-test='content']");
    expect(content).toExist();
  });
});
