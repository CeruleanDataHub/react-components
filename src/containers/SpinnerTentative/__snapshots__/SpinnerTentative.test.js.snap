// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerTentative renders 1`] = `
<SpinnerTentative
  condition={false}
>
  <styled.div>
    <div
      className="sc-AxjAm HfjjY"
    >
      <styled.div>
        <div
          className="sc-AxirZ cKyuOz"
        />
      </styled.div>
    </div>
  </styled.div>
</SpinnerTentative>
`;
